File: frontend/src/types/MonitorDTO.ts
// src/types/monitor.ts (or wherever you keep your DTO interfaces)

// Assuming your Java enums Monitor.MonitorType and Monitor.MonitorStatus
// are serialized as strings by your Spring Boot backend.
// These types should precisely match the enum values in your Java Monitor class.

export type MonitorType = 'HTTP_HTTPS' | 'PING' | 'PORT'; // Updated based on Java Monitor.MonitorType enum
export type MonitorStatus = 'UP' | 'DOWN' | 'PAUSED' | 'UNKNOWN'; // Updated based on Java Monitor.MonitorStatus enum

export interface MonitorDTO {
  id: number; // Corresponds to Java Long
  userId: number; // Corresponds to Java Long (from the 'user' relationship)
  url: string; // Corresponds to Java String
  checkIntervalSeconds: number; // Corresponds to Java Integer
  type: MonitorType; // Corresponds to Java Monitor.MonitorType enum
  status: MonitorStatus; // Corresponds to Java Monitor.MonitorStatus enum
  lastCheckedAt: string; // Corresponds to Java LocalDateTime, serialized as ISO 8601 string
  lastStatusChangeAt: string; // Corresponds to Java LocalDateTime, serialized as ISO 8601 string
  contentMatchString?: string; // New: Corresponds to Java String, optional as it's not nullable
  httpMethod?: string; // New: Corresponds to Java String, optional as it's not nullable
  httpHeaders?: string; // New: Corresponds to Java String (even if DB is jsonb, Java DTO is String), optional as it's not nullable
  createdAt: string; // Corresponds to Java LocalDateTime, serialized as ISO 8601 string
  updatedAt?: string; // New: Corresponds to Java LocalDateTime, serialized as ISO 8601 string, optional as it's not nullable
}

File: backend/src/main/java/com/watchdog/dto/MonitorDTO.java
package com.watchdog.dto;

import com.watchdog.entity.Monitor;

import java.time.LocalDateTime;


public class MonitorDTO {
    private Long id;
    private Long userId; // Include user ID for context
    private String url;
    private Integer checkIntervalSeconds;
    private Monitor.MonitorType type;
    private Monitor.MonitorStatus status;
    private LocalDateTime lastCheckedAt;
    private LocalDateTime lastStatusChangeAt;
    private LocalDateTime createdAt;

    public Monitor.MonitorStatus getStatus() {
        return status;
    }

    public void setStatus(Monitor.MonitorStatus status) {
        this.status = status;
    }

    public LocalDateTime getLastStatusChangeAt() {
        return lastStatusChangeAt;
    }

    public void setLastStatusChangeAt(LocalDateTime lastStatusChangeAt) {
        this.lastStatusChangeAt = lastStatusChangeAt;
    }

    public LocalDateTime getLastCheckedAt() {
        return lastCheckedAt;
    }

    public void setLastCheckedAt(LocalDateTime lastCheckedAt) {
        this.lastCheckedAt = lastCheckedAt;
    }

    public void setCreatedAt(LocalDateTime createdAt) {
        this.createdAt = createdAt;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setType(Monitor.MonitorType type) {
        this.type = type;
    }

    public Monitor.MonitorType getType() {
        return type;
    }

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public Long getUserId() {
        return userId;
    }

    public void setUserId(Long userId) {
        this.userId = userId;
    }

    public String getUrl() {
        return url;
    }

    public void setUrl(String url) {
        this.url = url;
    }

    public Integer getCheckIntervalSeconds() {
        return checkIntervalSeconds;
    }

    public void setCheckIntervalSeconds(Integer checkIntervalSeconds) {
        this.checkIntervalSeconds = checkIntervalSeconds;
    }
}